src/components/AgendaItem/AgendaItem.tsx

```diff
@@ -41,32 +41,32 @@ const AgendaItem: React.FC<AgendaItemProps> = ({
   };
 
   return (
-    <div
-      className="agenda-item"
-      style={{ backgroundColor, borderRadius, paddingRight }}
-    >
-      <div className="agenda-item__main">
-        <div className={"agenda-item__main__time"}>
-          <div>{startDate}</div>
-          <div>{endDate}</div>
-        </div>
-        <div className="agenda-item__main__content">
-          <Link to={`${ROUTES.TALKS}/${id}`}>
+    <Link to={`${ROUTES.TALKS}/${id}`}>
+      <div
+        className="agenda-item"
+        style={{ backgroundColor, borderRadius, paddingRight }}
+      >
+        <div className="agenda-item__main">
+          <div className={"agenda-item__main__time"}>
+            <div>{startDate}</div>
+            <div>{endDate}</div>
+          </div>
+          <div className="agenda-item__main__content">
             <div className="agenda-item__main__content__title">{title}</div>
-          </Link>
-          <div className="agenda-item__main__content__tagged">
-            {stage ? <Stage name={stage} /> : null}
-            {category ? (
-              <CategoryIndicator name={category || "no track"} />
-            ) : null}
+            <div className="agenda-item__main__content__tagged">
+              {stage ? <Stage name={stage} /> : null}
+              {category ? (
+                <CategoryIndicator name={category || "no track"} />
+              ) : null}
+            </div>
           </div>
         </div>
+        {/* use debounce if data is saved to swarm: debounce(handleClick, debounceTime) */}
```

Comment:
Heart icon also activates AgendaItem onclick.

Heart icon should not switch page, handleClick should contain e.preventDefault(), or something like that.